src/main/java/seedu/finance/model/record/Date.java

```diff
@@ -54,6 +55,9 @@ public static boolean isValidDate(String test) {
             int month = Integer.parseInt(parsedDate[1]);
             int day = Integer.parseInt(parsedDate[0]);
             LocalDate date = LocalDate.of(year, month, day);
+            if (date.isAfter(LocalDate.now())) {
+                return false;
+            }
         } catch (Exception e) {
```

Comment:
LocalDate has a isAfter method defined, so probably no need to write another method below

Reply:
I did not add that method so i did not change it or use it. I think @geezlouisee added it because it is used in the summary methods.